### ISSUE TITLE
Resume play doesn't work from Photo library videos

### ISSUE THREAD

**Issue by ohshitgorillas** [jellyfin-archive/jellyfin-roku-legacy#1624]
## Software Versions

- Jellyfin Server Version:  10.8.13 (Docker)
- Roku Client Version:  2.0 build 0

## Describe the bug
When trying to resume play from a previous position with videos in the Photo library, the video always starts from the beginning.

Videos in the Movies and Shows library always resume properly.

## How To Reproduce

1. Go to Photo library
2. Click on a video and watch it
3. Exit from the video
4. Try to resume the video

## Expected behavior
The video should resume play from its previous position, not from the beginning of the video.

## Connection Information

- Is server local or remote? Local
- Is server connection HTTP or HTTPS? HTTP

**Comment by dathbe** [comment]
How long are the videos?  If they are short, it will mark them as fully played when you play just a few seconds of them, and then it will play from the beginning when you restart.  Basically, if videos are short enough, there is no way to pause and resume in the middle because they will either not be playing long enough to set a bookmark or will be playing too long and consider play complete.  There is a setting to customize some of this behavior, but the shorter the video the harder it will be to set bookmarks.

**Comment by ohshitgorillas** [comment]
The videos range in length from 5 minutes to 3 hours. Most are in the 20-40 minute range and are stopped somewhere in the middle. Furthermore, the app asks me if I want to resume and I say "yes" but it plays from the beginning.

**Comment by ohshitgorillas** [comment]
Oops, didn't mean to close the issue, it's definitely a bug

**Comment by dathbe** [comment]
Hmm, that does sound like a bug. Have you checked that your resume settings aren’t set to something weird?  I don’t have a photos library set up (or 3hr home movies), so I can’t help much more.

**Comment by ohshitgorillas** [comment]
Yes, the resume settings are fine as they work with Movies and Shows libraries.

Actually, the timing of resumed videos in Movies and Shows is off as it's never where I left off, and when I resume it takes time for the audio and video to sync back up. But that's a different problem...

**Comment by dathbe** [comment]
The movies issues might be a transcoding issue. I’ve found that I can get sync issues with transcodes, so I go out of my way to make sure my files direct play on my usual clients.

**Comment by ohshitgorillas** [comment]
I've never had a transcoding issue like this before and I do a lot of hardware transcoding. That issue is unique to resuming playback.

**Comment by jimdogx** [comment]
Sorry, I should have chimed in earlier.  In all honestly, video support while viewing Photos was an after thought.  I added the Photo library support and one of the last things I did was also allow the user to play videos if they happen to have any videos mixed in with their photos.  But there was not a lot of effort in making it perform like a full video player.

I wasn't even sure anyone was even using Photo libraries on the Roku (never heard anyone say anything) until this issue 🙃 

This could be considered a bug or an enhancement.  Either way, I don't think it's anything you're doing wrong (e.g. transcoding, video length, etc)... it most likely just doesn't support video resume right now.

**Comment by ohshitgorillas** [comment]
Well, the video resume feature is already there... so I'm hoping it's a relatively simple task to make it work. It does ask me if I want to resume the video, it just fails to do so when I say yes. It did work in the previous version of the Roku app as well.

**Comment by dathbe** [comment]
@ohshitgorillas , what happens if you play a video past the resume point (I think default is 5-10% of the video), then back ALL the way out to the home screen, then go back in and try to play the video?  It turns out there is a bug with resuming all videos (#1647 ) where the save point is not updated on the video detail page right away.

**Comment by jsevison** [comment]
Server details:
Version: 10.8.13
Operating System: Linux
Architecture: X64

Client: 
TCL Roku TV
JellyFin Version 2.0 build 5

@ohshitgorillas , you reported that Videos in the Movies and Shows library always resume properly. It is my experience that this is NOT the case. 
 
Steps to Reproduce:
1. From a desktop device, logged in as user jellyfin, pause a movie. In my case movie progress is at 1:13:11
2. From Roku client, login as user jellyfin.
3. Press Continue Watching.
4. Press Play.
5. Playback Options correctly shows Resume playing at 1:13:12.
6. Press resume playing
7. Movie starts playing at beginning, rather than at 1:13:12.
8. Then I try to use fast forward. The slider bar progresses correctly. However, no matter where I attempt to start the movie at, it begins playing at the start of the movie.

Can someone comment on whether I should I create a new issue since this appears to be different behavior than reported with the existing issue?

**Comment by ohshitgorillas** [comment]
> @ohshitgorillas , what happens if you play a video past the resume point (I think default is 5-10% of the video), then back ALL the way out to the home screen, then go back in and try to play the video? It turns out there is a bug with resuming all videos (#1647 ) where the save point is not updated on the video detail page right away.

Sorry for neglecting to respond to this for so long.

The resume point is not saved even after exiting the app or returning to the home screen. The video still plays from the beginning rather than the resume point.
